pkg/operator/operator_config.go

```diff
@@ -352,20 +413,18 @@ func (r *operatorConfigReconciler) makeAlertManagerConfigs(ctx context.Context,
 	)
 
 	alertManagers := spec.Alertmanagers
-	if spec.EnableManagedAlertManager {
-		amNamespacedName := types.NamespacedName{
-			Namespace: r.opts.PublicNamespace,
-			Name:      "gmp-alertmanager",
-		}
 
-		// if the default AlertManager exists, append it to the list of spec.Alertmanagers
-		if resourceErr := r.client.Get(ctx, amNamespacedName, &appsv1.StatefulSet{}); resourceErr == nil {
-			alertManagers = append(alertManagers, monitoringv1.AlertmanagerEndpoints{
-				Name:      amNamespacedName.Name,
-				Namespace: amNamespacedName.Namespace,
-				Port:      intstr.FromInt(9093),
-			})
-		}
+	amNamespacedName := types.NamespacedName{
+		Namespace: r.opts.OperatorNamespace,
+		Name:      "alertmanager",
+	}
+	// if the default AlertManager exists, append it to the list of spec.Alertmanagers
+	if resourceErr := r.client.Get(ctx, amNamespacedName, &appsv1.StatefulSet{}); resourceErr == nil {
+		alertManagers = append(alertManagers, monitoringv1.AlertmanagerEndpoints{
+			Name:      amNamespacedName.Name,
+			Namespace: amNamespacedName.Namespace,
+			Port:      intstr.FromInt(9093),
```

Comment:
Fine for now but note the possibility that users may edit the alertmanager port in the yaml and break this. Can make it configurable if that use case does come up.

Reply:
What we should do is actually see if the default AM [_service_ ](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/8e89c873d3bc21f735941b968461c5e48c153d9f/cmd/operator/deploy/operator/12-alertmanager.yaml#L23-L35) exists, and use the port in the retrieved service object directly.

I can do that in a follow-up PR to shake off some of the cobwebs 🙂

Reply:
https://github.com/GoogleCloudPlatform/prometheus-engine/pull/314